include/swift/RemoteInspection/ReflectionContext.h

```diff
@@ -649,48 +674,96 @@ class ReflectionContext
 
     SwiftObjectFileFormatELF ObjectFileFormat;
     auto FieldMdSec = findELFSectionByName(
-        ObjectFileFormat.getSectionName(ReflectionSectionKind::fieldmd));
+        ObjectFileFormat.getSectionName(ReflectionSectionKind::fieldmd), true);
     auto AssocTySec = findELFSectionByName(
-        ObjectFileFormat.getSectionName(ReflectionSectionKind::assocty));
+        ObjectFileFormat.getSectionName(ReflectionSectionKind::assocty), true);
     auto BuiltinTySec = findELFSectionByName(
-        ObjectFileFormat.getSectionName(ReflectionSectionKind::builtin));
+        ObjectFileFormat.getSectionName(ReflectionSectionKind::builtin), true);
     auto CaptureSec = findELFSectionByName(
-        ObjectFileFormat.getSectionName(ReflectionSectionKind::capture));
+        ObjectFileFormat.getSectionName(ReflectionSectionKind::capture), true);
     auto TypeRefMdSec = findELFSectionByName(
-        ObjectFileFormat.getSectionName(ReflectionSectionKind::typeref));
+        ObjectFileFormat.getSectionName(ReflectionSectionKind::typeref), true);
     auto ReflStrMdSec = findELFSectionByName(
-        ObjectFileFormat.getSectionName(ReflectionSectionKind::reflstr));
+        ObjectFileFormat.getSectionName(ReflectionSectionKind::reflstr), true);
     auto ConformMdSec = findELFSectionByName(
-        ObjectFileFormat.getSectionName(ReflectionSectionKind::conform));
+        ObjectFileFormat.getSectionName(ReflectionSectionKind::conform), true);
     auto MPEnumMdSec = findELFSectionByName(
-        ObjectFileFormat.getSectionName(ReflectionSectionKind::mpenum));
+        ObjectFileFormat.getSectionName(ReflectionSectionKind::mpenum), true);
 
     if (Error)
       return false;
 
+    std::optional<uint32_t> result = false;
+
     // We succeed if at least one of the sections is present in the
     // ELF executable.
-    if (FieldMdSec.first == nullptr &&
-        AssocTySec.first == nullptr &&
-        BuiltinTySec.first == nullptr &&
-        CaptureSec.first == nullptr &&
-        TypeRefMdSec.first == nullptr &&
-        ReflStrMdSec.first == nullptr &&
-        ConformMdSec.first == nullptr &&
-        MPEnumMdSec.first == nullptr)
+    if (FieldMdSec.first != nullptr ||
+        AssocTySec.first != nullptr ||
+        BuiltinTySec.first != nullptr ||
+        CaptureSec.first != nullptr ||
+        TypeRefMdSec.first != nullptr ||
+        ReflStrMdSec.first != nullptr ||
+        ConformMdSec.first != nullptr ||
+        MPEnumMdSec.first != nullptr) {
```

Comment:
Personally, I would rely on integer promotion and avoid the ` != nullptr` as it just adds noise.

Reply:
:-) I was just keeping the existing style. (I would normally do exactly what you suggest.) Maybe we should change it.